data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/configuration/AWSAuthenticationOptions.java

```diff
@@ -23,5 +33,33 @@ public String getAwsRegion() {
     public String getAwsStsRoleArn() {
         return awsStsRoleArn;
     }
+
+    public AwsCredentialsProvider authenticateAWSConfiguration(final StsClient stsClient) {
+
+        AwsCredentialsProvider awsCredentialsProvider;
+        if (awsStsRoleArn != null && !awsStsRoleArn.isEmpty()) {
+            checkArgument(awsStsRoleArn.length() <= 2048, "awsStsRoleArn length cannot exceed 2048");
```

Comment:
Is this a hard requirement from AWS STS? If so, this check is good to have. But, we should also add a JSR validation to the property above as well. That will help give the user a clearer error sooner.

Reply:
You can use `@Size(max = 2048)` and JSR will only validate if it is not null. It will not require a non-null value. Also, would we want to set a minimum value? It looks like there might be a minimum of 20: https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html

Reply:
AWS ARN has length constrains.
Length Constraints: Minimum length of 1. Maximum length of 2048.

I'll add the JSR validation here instead of `checkArguemnt`.